packages/gatsby-core-utils/src/fetch-remote-file.ts

```diff
@@ -159,6 +159,24 @@ const requestRemoteNode = (
     })
   })
 
+const fetchCache = new Map()
+
+export function fetchRemoteFileWithCache(
+  args: IFetchRemoteFileOptions
+): Promise<string> {
+  // If we are already fetching the file, return the unresolved promise
+  const inFlight = fetchCache.get(args.url)
+  if (inFlight) {
+    return inFlight
+  }
+
+  // Create file fetch promise and store it into cache
+  const fetchPromise = fetchRemoteFile(args)
+  fetchCache.set(args.url, fetchPromise)
+
+  return fetchPromise 
+}
+
```

Comment:
Yes, please move this to the original fetch

Reply:
Ill do so :)

Reply:
Done!